app/processing/sfi-23-advanced-statement/get-sfi-23-advanced-statement.js

```diff
@@ -0,0 +1,43 @@
+const db = require('../../data')
+const { getDetails, getAddress, getScheme, getDetailedPayments } = require('../components')
+const { getLatestInProgressPaymentRequest } = require('../payment-request')
+const { getSettlement, getSupportingSettlements } = require('../settlement')
+const { getLatestPayment } = require('../payment')
+const getPaymentRequestByPaymentRequestId = require('../../inbound/get-payment-request-by-payment-request-id')
+const getOrganisationByFrn = require('../organisation/get-organisation-by-frn')
+const sfiaGetFunding = require('../components/sfia-hard-coded/sfia-fundings')
+
+const getSfi23AdvancedStatement = async (settlementId, scheduleId) => {
+  const transaction = await db.sequelize.transaction()
+  try {
+    const settlement = await getSettlement(settlementId, transaction)
+    const paymentRequest = await getLatestInProgressPaymentRequest(settlement.paymentRequestId, settlement.settlementDate, transaction)
+    const sfiaPaymentRequest = await getPaymentRequestByPaymentRequestId(paymentRequest.paymentRequestId)
+    const sfiaOrganisation = await getOrganisationByFrn(sfiaPaymentRequest.frn)
+    const sfiaCalculation = { calculated: '05/11/2023', sbi: sfiaOrganisation.sbi }

```

Comment:
all sfi advanced will have the same calculated date?

Reply:
Yes, they all have the same calculation date, also, calculation date is not required for sfi-23-advanced statement, but we have to provide it as it is a mandatory field for statement schema, which is the base object for sfi-23-advanced-statement.